src/net.h

```diff
@@ -421,8 +500,10 @@ class CNode
 
     const ConnectionType m_conn_type;
 
-    /** Move all messages from the received queue to the processing queue. */
-    void MarkReceivedMsgsForProcessing()
+    /** Move all messages from the received queue to the processing queue.
+     * Also update the global map of network message statistics.
+     */
+    void MarkReceivedMsgsForProcessing(NetStats& net_stats)
```

Comment:
Would it be better to pass a reference of NetStats to every CNode upon initialization?

Example: https://github.com/bitcoin/bitcoin/commit/cd0c8eeb0940790b6ba83786d1c9e362d4dc4829

Reply:
According to this review comment: https://github.com/bitcoin/bitcoin/pull/27257#discussion_r1146305999 - It looks like it would be preferable for CNode to have net_stats as a member variable (I could be wrong) but that might mean each CNode would need to have a shared_ptr to a NetStats instance.

Reply:
This is a great point. I think I need to take a closer look at some of the refactoring that's going on in `net` in hopes of finding other examples/decisions that support making `NetStats` a member variable in `CNode`. I think the comment you linked to is very relevant.